core/src/test/java/io/confluent/kafka/schemaregistry/ClusterTestHarness.java

```diff
@@ -79,59 +104,41 @@ public ClusterTestHarness(int numBrokers, boolean setupRestApp) {
   }
 
   public ClusterTestHarness(int numBrokers, boolean setupRestApp, String compatibilityType) {
-    // 1 port for ZK, 1 port per broker, and 1 port for the rest App if needed
-    int numPorts = 1 + numBrokers + (setupRestApp ? 1 : 0);
+    this.numBrokers = numBrokers;
+    this.setupRestApp = setupRestApp;
+    this.compatibilityType = compatibilityType;
+  }
 
-    ports = new ArrayDeque<Integer>();
-    for (Object portObj : JavaConversions.asJavaList(TestUtils.choosePorts(numPorts))) {
-      ports.add((Integer) portObj);
-    }
-    zkPort = ports.remove();
-    zkConnect = String.format("127.0.0.1:%d", zkPort);
+  @Before
+  public void setUp() throws Exception {
+    zookeeper = new EmbeddedZookeeper();
+    zkConnect = String.format("127.0.0.1:%d", zookeeper.port());
+    zkClient =
+        new ZkClient(zkConnect, zkSessionTimeout, zkConnectionTimeout,
+                     ZKStringSerializer$.MODULE$);
 
-    configs = new Vector<KafkaConfig>();
-    bootstrapServers = "";
+    configs = new Vector<>();
+    servers = new Vector<>();
     for (int i = 0; i < numBrokers; i++) {
-      int port = ports.remove();
-      Properties props = TestUtils.createBrokerConfig(i, port, false);
+      final Option<java.io.File> noFile = scala.Option.apply(null);
+      Properties props = TestUtils.createBrokerConfig(i, zkConnect, false, false, TestUtils.RandomPort(), false, TestUtils.RandomPort(), noFile);
```

Comment:
Ok nm - my understanding is that this sets the port config to 0, and the brokerListStr below actually pulls the port from the socket when constructing the string